Toolkit-API/Device/HardwareInfo.cs

```diff
@@ -0,0 +1,102 @@
+
+            JObject messageObj;
+            try {
+                messageObj = JObject.Parse(message);
+            }
+            catch (JsonReaderException ex) {
+                // Handle invalid JSON
+                Console.WriteLine("Invalid JSON string: " + ex.Message);
+                return hardwareInfoList;
+            }
+
+            // Navigate to the "payload" -> "value"
+            if (messageObj.TryGetValue("payload", out JToken payloadToken) &&
+                payloadToken is JObject payloadObj &&
+                payloadObj.TryGetValue("value", out JToken payloadValueToken) &&
+                payloadValueToken is JObject payloadValueObj) {
+
+                // Iterate over each hardware item
+                foreach (var item in payloadValueObj.Properties()) {
+                    if (item.Value is JObject itemObj &&
+                        itemObj.TryGetValue("value", out JToken itemValueToken) &&
+                        itemValueToken is JObject itemValueObj) {
+
+                        HardwareInfo hardwareInfo = new();
+
+                        // Parse each property without extension method
+                        hardwareInfo.index = GetValueFromProperty<int>(itemValueObj, "index");
+                        hardwareInfo.hardwareVersion = GetValueFromProperty<string>(itemValueObj, "hardwareVersion");
+                        hardwareInfo.hardwareVersionLong = GetValueFromProperty<string>(itemValueObj, "hardwareVersionLong");
+                        hardwareInfo.hasEdidCalibration = GetValueFromProperty<bool>(itemValueObj, "hasEdidCalibration");
+                        hardwareInfo.hfov = GetValueFromProperty<float>(itemValueObj, "hfov");
+                        hardwareInfo.vfov = GetValueFromProperty<float>(itemValueObj, "vfov");
+                        hardwareInfo.viewCone = GetValueFromProperty<float>(itemValueObj, "viewCone");
+                        hardwareInfo.resolutionWidth = GetValueFromProperty<int>(itemValueObj, "resolutionWidth");
+                        hardwareInfo.resolutionHeight = GetValueFromProperty<int>(itemValueObj, "resolutionHeight");
+                        hardwareInfo.defaultQuilt = GetValueFromProperty<string>(itemValueObj, "defaultQuilt");
+                        hardwareInfo.calibration = GetValueFromProperty<string>(itemValueObj, "calibration");
+
+                        hardwareInfoList.Add(hardwareInfo);
+                    }
+                }
+            }
+
+            return hardwareInfoList;
+        }
+
+        // Helper method inside the struct
+        private static T GetValueFromProperty<T>(JObject obj, string propertyName) {
+            if (obj.TryGetValue(propertyName, out JToken token) &&
+                token is JObject valueObj &&
+                valueObj.TryGetValue("value", out JToken valueToken)) {
+
+                return valueToken.ToObject<T>();
+            }
+            return default;
+        }
```

Comment:
Oh here we can actually delete this helper method and use the JObjectExtensions's TryGet<T>(...) instead if you'd like, since they do the same thing! Example usage in Calibration.cs:

![image](https://github.com/user-attachments/assets/1f916973-cd4a-4de8-87b2-cb2296b3a646)

Reply:
Great works for me, I just dont ever use newtonsoft json so I missed this